plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -391,6 +391,11 @@ protected function status_transition() {
 		$this->status_transition = false;
 
 		if ( $status_transition ) {
+			// Verify planned status transition did happen, in case save() wasn't successful.
+			if ( $status_transition['to'] !== $this->data['status'] ) {
```

Comment:
let's use get_changes instead, generally, we don't access the data array directly outside of getters/setters.
```suggestion
			$changes = $this->get_changes();
			if ( isset( $changes['status'] ) && $changes['status'] === $status_transition['to'] ) {
```

Reply:
Here, `$changes['status']` has the "changed value" ("refunded" in our exemple) but `$this->data['status']` still holds the database persisted value ("pending" in our exemple).
![image](https://github.com/user-attachments/assets/0e7c27be-c080-4128-a862-d76835369f4a)
We want to check the `$status_transition['to']` value against the current persisted value `$this->data['status']` and therefore cannot use the `get_changes()` method.

Reply:
We could maybe  use `$this->get_base_data()` if we prefer to use a getter ?

Reply:
Here, `$this->get_changes()['status']` has the "changed value" ("refunded" in our exemple) but `$this->data['status']` still holds the database persisted value ("pending" in our exemple).
![image](https://github.com/user-attachments/assets/0e7c27be-c080-4128-a862-d76835369f4a)
We want to check the `$status_transition['to']` value against the current persisted value `$this->data['status']` and therefore cannot use the `get_changes()` method.

We could maybe  use `$this->get_base_data()` if we prefer to use a getter ?

Reply:
@vedanshujain can you have a look at my comments above ?